src/Groups/matrices/iso_oscar_gap.jl

```diff
@@ -0,0 +1,118 @@
+   codomain::GapObj          # GAP field
+   f                      # function from the Oscar field to the GAP field
+   f_inv               # its inverse
+end
+
+# isomorphism from the Oscar matrix space to the GAP matrix space
+struct GenMatIso <: Map{FqNmodMatSpace,GapObj,SetMap,GenMatIso}
+   domain::FqNmodMatSpace        # Oscar matrix space
+   codomain::GapObj             # GAP matrix space
+   fr::GenRingIso             # see type above
+   f                         # function from the Oscar matrix space to the GAP matrix space
+   f_inv                  # its inverse
+end
+
+
+################################################################################
+#
+#  Ring isomorphism
+#
+################################################################################
+
+# return the GAP element corresponding to the Oscar element x
+function elem_f(x::fq_nmod, B,d)
+   COEF = [Int64(coeff(x,i)) for i in 0:d-1]
+   x_gap = sum([COEF[i]*B[i] for i in 1:d])
+   return x_gap
+end
+
+# return the Oscar element corresponding to the GAP element x
+function elem_g(x::FFE, B, z, d)
+   L = Vector{Int}(GAP.Globals.List(GAP.Globals.Coefficients(B,x),GAP.Globals.IntFFE ))
+   return sum([L[i]*z^(i-1) for i in 1:d])
+end
+
+
+# computes the isomorphism between the Oscar field F and the corresponding GAP field F_gap
+# the output has type GenRingIso (see above)
+
+function gen_ring_iso(F::FqNmodFiniteField)
+   p = Int64(characteristic(F))
+   d = Int64(degree(F))
+   z = gen(F)
+
+   if d==1
+      myOne = GAP.Globals.One(GAP.Globals.GF(p))
+      f(x::fq_nmod) = Int(coeff(x,0))*myOne
+      finv(x::FFE) = F(GAP.Globals.IntFFE(x))
+      return GenRingIso(F, GAP.Globals.GF(p), f, finv)
+   end
+   L = [Int64(lift(coeff(defining_polynomial(F),i))) for i in 0:d-1]
+   L = vcat(L,[1])
+   i = GAP.Globals.One(GAP.Globals.GF(p))
+   L_gap = GAP.julia_to_gap([i*y for y in L])
+   f_gap = GAP.Globals.UnivariatePolynomial(GAP.Globals.GF(p),L_gap)
+   F_gap = GAP.Globals.GF(GAP.Globals.GF(p),f_gap)
+   Basis_F = GAP.Globals.Basis(F_gap)
+   homom(x::fq_nmod) = elem_f(x,GAP.gap_to_julia(GAP.Globals.BasisVectors(Basis_F)),d)
+   homominv(x::FFE) = elem_g(x, Basis_F, z, d)
+   return GenRingIso(F, F_gap, homom, homominv)
+end
+
+(g::GenRingIso)(x::fq_nmod) = g.f(x)
+(g::GenRingIso)(x::FFE) = g.f_inv(x)
+
+Base.show(io::IO, f::GenRingIso) = print(io, "Ring isomorphism between ", f.domain, " and the corresponding GAP")
+
+
+
+################################################################################
+#
+#  Matrix space isomorphism
+#
+################################################################################
+
+# return the GAP matrix corresponding to the Oscar matrix x
+# assumes x is a square matrix
+function mat_oscar_gap(x::fq_nmod_mat, riso)
+   n = nrows(x)
+   S = Vector{GapObj}(undef, n)
+   for i in 1:n
+      S[i] = GAP.julia_to_gap([riso(x[i,j]) for j in 1:n])
+   end
+
+   return GAP.julia_to_gap(S)
+end
```

Comment:
This function returns one possible matrix object, but perhaps there are better ones.
For example, the elements of GAP's `GL(n,q)` (for small `q`) are "compressed" matrices.
I am not sure what would be best to achieve nicer matrices here. Either we have to provide a function in GAP.jl that applies some heuristics, or the object `riso` must know about the intended GAP type of the result.

Reply:
I just used the standard GAP constructor for matrices. Anyway, the operations involving GAP procedures should deal only with elements of `GL(n,q)`, so if it's possible to write a "compressed" matrix, that would be fine.

Reply:
I would write `return GAP.Globals.ImmutableMatrix( F, GAP.julia_to_gap(S), true)` where `F` is the GAP field from `riso` hich contains the matrix entries. (Note that this field may be larger than the field generated by the matrix entries.)

Reply:
Done it.

Reply:
There are many further optimizations possible here, but they require deeper level access to Nemo/Flint matrices; I'll add some suggestions in an issue later

Reply:
Yes.
My idea behind proposing the `ImmutableMatrix` call here was that the GAP object obtained this way is **what** one wants on the GAP side: a matrix that knows the context `F` (if possible). Improvements **how** one gets such a GAP object are of course welcome.